src/hypothesis/extra/numpy.py

```diff
@@ -56,12 +72,19 @@ class ArrayStrategy(SearchStrategy):
     def __init__(self, element_strategy, shape, dtype):
         self.shape = tuple(shape)
         assert shape
-        self.array_size = reduce(operator.mul, shape)
+        self.array_size = np.prod(shape)
+        if self.array_size * dtype.itemsize > settings().buffer_size:
```

Comment:
Hmm, what I really want to do is get the size of the current buffer - you're saying that `settings.default.buffer_size` reflects any user settings too?